course_discovery/apps/course_metadata/models.py

```diff
@@ -1590,17 +1592,17 @@ def search(cls, query):
         Args:
             query (str) -- Elasticsearch querystring (e.g. `title:intro*`)
         Returns:
-            SearchQuerySet
+            Search object
         """
         query = clean_query(query)
-        queryset = SearchQuerySet().models(cls)
-
+        es_document, *_ = registry.get_documents(models=(cls,))
+        queryset = es_document.search()
         if query == '(*)':
             # Early-exit optimization. Wildcard searching is very expensive in elasticsearch. And since we just
-            # want everything, we don't need to actually query elasticsearch at all.
-            return queryset.load_all()
+            # want everything, we don't need to actually query elasticsearch at all
+            return queryset.query(ESDSLQ('match_all'))
```

Comment:
I forgot about the comment above, apparently, I missed it.
Everything is not so simple here and it is impossible to access the database if `(*)` without additional development will not work. I assume that exactly the request with `(*)` when https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html is so heavy and expensive, I believe that `{'query': {'match_all': {}}` will not be so expensive that es7 behaves better than es1. This can only be verified in real-life mode and understand how difficult it is.

Why can't you just refer to the SQL database like `cls.objects.all ()`?
The problem is that we need to support the queryset protocol here for elasticsearch-dsl and it is different from the django ORM protocol. Further es queryset will need to be filtered based on the elasticsearch-dsl protocol. Here's an example to show what I mean.
https://github.com/raccoongang/course-discovery/blob/ddac9108ccfb71b669c606460991976b86e4b278/course_discovery/apps/api/v1/views/course_runs.py#L95

Also, the request itself is wrapped in a `SearchQuerySetWrapper` wrapper that expects the eleasticsearch dsl protocol. In general, a lot of problems 🙂

`queryset.load_all()` - this was exactly a feature of django-hastack, this library caches a lot of data from the SQL database, and can go to the database, not es. django-elasticsearch-dsl simply doesn't have this.

Here we have two options in my opinion:
1. See how `{'query': {'match_all': {}}` will work in real conditions.
2.Develop a wrapper for django-elasticsearch-dsl to do what django-hastack `queryset.load_all ()` does.

Reply:
I'm down for trying `match_all`. If I recall correctly, the reason `(*)` was so slow was that it was regex searching all possible ES keys or something similarly crazy. `match_all` seems like a straightforward request to the ES backend that wouldn't do a crazy thing.

So here's hoping.

Reply:
Yeah, let's keep eyes and 🤞